main.go

```diff
@@ -423,6 +423,20 @@ func main() { //nolint:funlen,maintidx,gocyclo
 			os.Exit(1)
 		}
 	}
+
+	// TODO: to be removed: https://issues.redhat.com/browse/RHOAIENG-21080
+	var patchODCFunc manager.RunnableFunc = func(ctx context.Context) error {
+		if err := upgrade.PatchOdhDashboardConfig(ctx, setupClient, oldReleaseVersion, release); err != nil {
+			setupLog.Error(err, "Unable to patch the odhdashboardconfig")
+		}
+		return err
```

Comment:
what is the intent here ? the error returned bu the function is the one in the outer scope, if you want to return the error  from the `PatchOdhDashboardConfig` functions, you need to do something like:

```go
err := upgrade.PatchOdhDashboardConfig(...)
if err != nil {
    ....
    return err
}
```

Reply:
changed this